prog/weaveutil/datapath.go

```diff
@@ -21,7 +41,48 @@ func createDatapath(args []string) error {
 		// status to distinguish it for the weave script.
 		os.Exit(17)
 	}
-	return err
+	if err != nil {
+		return err
+	}
+
+	if setMTU {
+		if err := wnet.SetMTU(dpName, mtu); err != nil {
+			return fmt.Errorf("unable to set MTU to %d on %q interface: %s", mtu, dpName, err)
+		}
+		// Check whether we are able to set the given MTU on a vxlan vport in order
+		// to warn users exposed to https://github.com/weaveworks/weave/issues/1853
+		if mtu > 1500 {
+			if vpid, vpname, err := odp.CreateVxlanVport(dpName, "vxlan", getUDPPortNo()); err == nil {
+				defer func() {
+					if err := odp.DeleteVport(dpName, vpid); err != nil {
+						fmt.Fprintf(os.Stderr, "unable to remove unused %q vxlan vport: %s\n", vpname, err)
+					}
+				}()
+				if err := wnet.SetMTU(vpname, mtu); err != nil {
+					if errno, ok := err.(syscall.Errno); ok && errno == syscall.EINVAL {
+						fmt.Fprintf(os.Stderr, "WARNING: fastdp mode is not able to work with %d MTU potentially due to 4.3 or 4.4 kernel version\n", mtu)
```

Comment:
As I mentioned on slack, this really should tell the user what MTU *is* being used. Otherwise the warning might be misinterpreted as indicating that fastdp isn't working *at all*.